source/features/clone-branch.tsx

```diff
@@ -0,0 +1,69 @@
+import gitBranch from 'octicon/git-branch.svg';
+import React from 'dom-chef';
+import select from 'select-dom';
+import delegate, {DelegateEvent} from 'delegate-it';
+import * as api from '../libs/api';
+import features from '../libs/features';
+import {getCleanPathname} from '../libs/utils';
+
+async function cloneBranch(event: DelegateEvent<MouseEvent, HTMLAnchorElement>): Promise<void|false> {
+	const currentTarget = event.delegateTarget;
+	const newBranchName = prompt('Enter the new branch name');
+	if (!newBranchName) {
+		return false;
+	}
+
+	const spinner = currentTarget.nextElementSibling!.nextElementSibling! as HTMLAnchorElement;
+	spinner.hidden = false;
+	currentTarget.hidden = true;
+
+	const [user, repo] = getCleanPathname().split('/');
+	const {branchName} = (currentTarget.closest('[data-branch-name]') as HTMLAnchorElement).dataset;
+	const getBranchInfo = await api.v3(`repos/${user}/${repo}/git/refs/heads/${branchName!}`);
+	try {
+		await api.v3(`repos/${user}/${repo}/git/refs`, {
+			method: 'POST',
+			body: {
+				sha: String(getBranchInfo.object.sha),
+				ref: 'refs/heads/' + newBranchName
+			}
+		});
+		location.reload();
```

Comment:
I think there should be a success alert

Reply:
Make the new branch look selected on the next load, via `hash` if possible

Reply:
@fregante but the branch is not always visible. I can try, I just stepped away from the computer.

Reply:
Unless you can clone the whole branch DOM and change its name (if all the buttons within it work)

Reply:
Wow. You would rather that, then a simple alert?

Reply:
Definitely no alerts anywhere unless they’re errors

Reply:
Ok

Reply:
Looked into it.
if we cone the node we need to update the pull request and the delete buttons. (which I dont event know if it would work)
Should I leave them out?

Reply:
> which I dont event know if it would work

That's something that needs to be tested, if you want to send PRs. PRs are meant to reduce workload of maintainer, not increase it.

Reply:
The answer is no, the DOM has too much information and state, I thought we just needed to update branch name and delete button.